rts/at/bridge/src/main/java/org/jboss/narayana/rest/bridge/inbound/InboundBridgeManager.java

```diff
@@ -45,13 +45,9 @@
 
     private final Map<String, Xid> transactionXids;
 
-    public static InboundBridgeManager getInstance() {
+    public static synchronized InboundBridgeManager getInstance() {
```

Comment:
If you check the linked explanation you will see that the idiom is broken. Also, any performance gains made by implementing a more complex fix to avoid the synchronization overhead are questionable.